build_runner/lib/src/package_graph/package_graph.dart

```diff
@@ -179,6 +183,13 @@ class PackageNode {
 
   PackageNode(this.name, this.version, this.dependencyType, this.path);
 
+  /// Create a [PackageNode] without any details from a `pubspec.yaml` file.
+  ///
+  /// This is useful for testing, or in cases where a package may be synthetic.
+  PackageNode.noPubspec(this.name, this.path)
```

Comment:
Should we make path be a named optional arg as well?  It looks like we still pass null a fair bit.

Reply:
Sg. Done.